### ISSUE TITLE
Add custom Braille landmarks

### ISSUE THREAD

**Issue by vincent-lg** [aaclause/BrailleExtender#17]
I propose that, instead of using dot 78 to mark a formatting change, Braille delimiters be used to clearly identify different formats.

## Justification

Adding dot 7 and 8 to inform about a specific formatting offers several drawbacks:

- It drowns out these dots, and important information can sometimes be found on dots 7 and 8.  In most computer Braille tables, dot 7 is used to indicate capital letters.  While this information is not vital, not seeing it can be problematic and force the user to turn off Attributa just to examine a word.
- It doesn't allow to differentiate between different formats.

As it is, the user relying only on Braille (as I do) would turn on only one of the options of Attributa to avoid clustering and hard-to-decode information.  For instance, I like to see my spelling errors with dot 7 and 8, but if I turn anything else on, I won't know what's what.

## Proposal

Instead of using dot 7 and 8 to indicate these formatting changes, I propose to use clear Braille landmarks that, maybe, the user can choose.  There would be one Braille landmark before the affected portion, and another one at the end.  For instance:

    This is in <b>bold<B>.

I know this will add quite a layer of complexity, since it will shift Braille cells further and render NVDA's calculation for the routing cursor invalid, but I believe this is worth it.

I have detailed this proposal on a [NVDA issue](https://github.com/nvaccess/nvda/issues/8436) but have been asked to talk to you about the problem first.

**Comment by Andre9642** [comment]
Hello,
Thank you for this great suggestion. I also have this idea since a while.
I hope to implement it soon!

**Comment by Andre9642** [comment]
Yes, I'll try to implement this for the next version. I'd like to release a stable version for early August.

**Comment by Andre9642** [comment]
Hi,
I'd like to integrate this feature for January.
@vincent-lg, @zstanecic: if a piece of text is in bold and underline for example, should we combine the landmarks like this?
⣇⠃⠥⣸blah-blahh-blah⣇⡃⡥⣸
Also, if we click under '⣇⠃⠥⣸' or '⣇⡃⡥⣸' using routing cursor buttons, what behavior would you like?

**Comment by bdorer** [comment]
In my case, I'd like the Cursor jumps to the nearest char it the landmark belongs to. What do others think?

**Comment by vincent-lg** [comment]
Hi,
Sorry answering took so long.  I do agree with @bdorer.  It would make sense if clicking on the landmark jumped to the first character in the landmark.  NVDA behaves like this for other things, I believe, like HTML landmarks (ln, edit and so on).
Hope that helps,
Vincent

**Comment by aaclause** [comment]
@vincent-lg, @bdorer  Thank you for your feedback. I agree with you. I plan to resume development soon (this summer).
